### PR TITLE
Authorize and class attributes

### PR DESCRIPTION
**Pull request by kookster**
- [x] Singleton method for authorize not being created correctly, or explicitly returning `true` when not implemented
- [x] Several class attributes are being added in included modules, but not defined as inheritable `class_attribute`s
- [x] The hal api controller is a concern, but includes modules that are not, making the module larger, and preventing easy use of the `included` method for nested modules

### CODE REVIEW

**Comment by kookster** [comment]
(Ran across these issues while working on CMS)

File: lib/hal_api/controller.rb [hunk 1]
```diff
@@ -24,13 +24,6 @@ module HalApi::Controller
     hal_rescue_standard_errors
   end
 
-  module ClassMethods
-    include HalApi::Controller::Actions::ClassMethods
```
Comment (cavis) [inline lib/hal_api/controller.rb]: So you're thinking that now the sub-controller will include these concerns?

Comment (cavis) [inline lib/hal_api/controller.rb]: Oh wait, nevermind ... I seed that they're already included above this diff.

File: lib/hal_api/controller/actions.rb [hunk 2]
```diff
@@ -80,20 +88,18 @@ def root_resource(resource)
...
-      define_singleton_method(:authorize, Proc.new(resource) do
+    if !respond_to?(:authorize)
+      define_singleton_method(:authorize) do |_resource|
         true
-      end)
+      end
       singleton_class.send(:alias_method, :hal_authorize, :authorize)
     end
+
+    authorize(resource)
```
Comment (cavis) [inline lib/hal_api/controller/actions.rb]: Was the issue here that `hal_authorize` wasn't returning true on the first call, if you didn't explicitly define an `authorize`?  (It looks like it was returning the result of `alias_method` on the first call).

File: lib/hal_api/controller/actions.rb [hunk 3]
```diff
@@ -80,20 +88,18 @@ def root_resource(resource)
   end
 
   def hal_authorize(resource)
-    if respond_to?(:authorize)
-      authorize(resource)
-    else
-      define_singleton_method(:authorize, Proc.new(resource) do
```
Comment (kookster) [inline lib/hal_api/controller/actions.rb]: this was the bad syntax @cavis

**Pull request merged**